app/scripts/ui.js

```diff
@@ -348,7 +349,8 @@ function connectToAccountManager(connectionStream) {
     controllerConnectionStream,
   );
 
-  setupWeb3Connection(mx.createStream('provider'));
+  const providerStream = createDuplexStreamWrapper(mx, 'provider');
```

Comment:
Can you explain why this is needed? `mx.createStream('provider')` already returns a Duplex, and StreamProvider similarly already takes a Duplex.

Reply:
Gotcha, makes sense!